### DISCUSSION TITLE
Custom CPU and GPU temperature limits (currently 75 is min temp)

### DISCUSSION THREAD

**Discussion by Zackeroo** [seerge/g-helper#1475] [Ideas]
Universal x86 Tuning Utility (previously known as AATU) used to have the ability to set a custom max temperature, and this worked flawlessly. 

In hot conditions, having the laptop cool to the touch is excellent, and being able to lower the temperatures and throttle the laptop down is useful when pairing this with changing the fan curves too.

**Comment by AudreyAP** [comment]
This is already implemented for AMD CPUs and Nvidia GPUs. IIRC Intel has no documented endpoint for it, nor does AMD for their GPUs.

**Reply by Zackeroo** [reply]
This is precisely why I'm recommending it. Since it's already implemented, it should be quite straightforward to make this change to GHelper. Very odd decision to limit us to 75 degrees minimum within GHelper.

**Reply by AudreyAP** [reply]
I misread your comment - thought you were asking for custom temperature limits, period.

**Comment by seerge** [comment]
@Zackeroo hello, I'm intentionally not allowing "extreme" temp limits in app UI :) as this is just a throttle temperature (i.e. when your CPU will significantly drop power in order to cool down). As otherwise, people will set something like 50C in limit, and then go crying on redding that they computer is "lagging with g-helper" cause it runs with 5w limits ...

If your CPU goes hotter than 75C in idle / low load - there is something wrong with hardware.

**Reply by Zackeroo** [reply]
Yup I figured this would be the reason why you had 75 as the lowest available option. I completely understand your point here. 

That being said, why not just show a warning message when the slider is lowered below 75 degrees? "Decreasing your temperature limit below 75C will negatively impact performance". A little prompt making the user click "I Understand" could work too. 

**Alternate Solution** 
Realistically though, considering the style and approach you've gone for with GHelper, a toggle in settings would be a perfect fit for GHelper:
"Experimental: "Extended Temperature Limits"

Users will now know that this feature is "Experimental". This could even be paired with a little warning message to remind users that the feature is toggled. 

Your call. This is one of the only features making me have to the Universal x86 Tuning Utility.

**Reply by seerge** [reply]
@Zackeroo you can override min-temp value by adding

```
"min_temp" : 50,
```

To ``config.json`` 

As app already expects this value from config https://github.com/seerge/g-helper/blob/c978c94d59e58da3f5cf2adb9b52d93167799fec/app/Ryzen/RyzenControl.cs#L20

**Reply by moritzbeck1** [reply]
@seerge Is there an equivalent config for GPUs? I can't find any documentation on this, not even [here](https://github.com/seerge/g-helper/wiki/Power-user-settings). Is there a way to set this per GPU for devices with iGPU/dGPU/eGPU?

**Reply by seerge** [reply]
@moritzbeck1 GPU temp limit is set via Asus endpoint, and it can be only in a range app already offers you
